simulator/include/api.h

```diff
@@ -0,0 +1,40 @@
+#ifndef CHS_API_H
+#define CHS_API_H
+
+#include "system.h"
+
+namespace CacheHierarchySimulator
+{
+
+struct Instruction
+{
+    uint8_t read_write = 0;
+    uint64_t address = 0;
+    uint64_t cycleTime = 0;
+};
+
+/**
+ * Api for adding systems and running simulations on a list of instructions.
+ * Has mutiple systems to allow for system comparisons
+ * Will return a list of structs containing simulation results.
+*/
+class Api
+{
+public:
+
+    Api();
+    ~Api();
+
+    void addSystem(const System& system);
```

Comment:
Question: Is the idea here that you can simulate multiple cache structures at the same time?

Reply:
Yes, one function call to simulate the same instructions over multiple archs. I could see the argument for it being redundant, though.

Reply:
I like the idea of simulating multiple cache hierarchies simultaneously. Could prevent the user from running the simulator a bunch of times for each configuration.